packages/wrangler/src/config/validation.ts

```diff
@@ -111,6 +111,14 @@ export function normalizeAndValidateConfig(
     "boolean"
   );
 
+  validateOptionalProperty(
+    diagnostics,
+    "",
+    "node_compat",
+    rawConfig.node_compat,
+    "boolean"
+  );
```

Comment:
Indirectly related to this PR, however, I was thinking we could add a optional messaging to the validation framework. 

That would eliminate the need for things like 
```ts 
      if (nodeCompat) {
        logger.warn(
          "Enabling node.js compatibility mode for builtins and globals. This is experimental and has serious tradeoffs. Please see https://github.com/ionic-team/rollup-plugin-node-polyfills/ for more details."
        );
      }
      ```